### DISCUSSION TITLE
Enable If/Else to connect to Conversational QA for complex RAG setups

### DISCUSSION THREAD

**Discussion by themeaningofmeaning** [FlowiseAI/Flowise#1614] [Ideas]
It would be great if the If/Else Function could connect directly to the conversational QA chain.  I built out a RAG via prompt chaining (the if/else essentially determines intent before directing the conversation in the appropriate direction.  A direct If/Else connection to a conversational QA chain would eliminate six elements + 3 chains from something like this:
<img width="1658" alt="Screenshot 2024-01-25 at 5 17 55 PM" src="https://github.com/FlowiseAI/Flowise/assets/5422554/d2c6740d-8c4e-42b4-8b9b-e5990d79d906">

...feels like overkill.

**Comment by paulf12345** [comment]
Have you tried connecting agent actions to different vector DBs and then implementing the if/else logic into the prompt? An explanation of your use case might help improve people's ability to help.  I agree though that would be a sweet feature and allow for more agents to be used in conjunction with if/else RAG logic reliably.

**Reply by themeaningofmeaning** [reply]
I tried to use agent actions but the results have been inconsistent for some reason.  I'm using LangSmith to debug, but haven't figured out why agent actions occasionally default to the wrong QA prompt chain.  

The use case is essentially determining if a user is subtly hinting to 'end the subject focus of the current chat session' and would like a summary generated (and the subject focus 'reset' for the next conversation with a new unique conversation ID)...versus continuing the conversation while storing message history under that specific subject focus.  Each subject focus has a unique conversation ID to be stored in Zep/Oracle/etc that can then be loaded into a flow later on.

**Comment by mahmoudsorf** [comment]
@themeaningofmeaning can you export and share your workflow. It should be in Flowise Marketplace :)

**Comment by somia20** [comment]
I am trying to do RAG with agent in ifelse
<img width="762" alt="image" src="https://github.com/user-attachments/assets/1754fbad-6aef-47b3-8ac6-780e2da10f39">

though it is working for RAG part but for greeting part it is showing me this error
Error buildAgentGraph - Cannot read properties of undefined (reading 'name')
can anyone help me this!?
